agogosml_cli/cli/generate.py

```diff
@@ -132,6 +136,16 @@ def extract_azure_template_vars(manifest: dict) -> dict:
     return template_vars
 
 
+def add_agogosml_template_vars() -> dict:
+    """Add agogosml-specific template variables"""
+    vinfo = VersionInfo.parse(__version__)
+    return {
+        'AGOGOSML_CLI_VERSION': __version__,
+        'AGOGOSML_CLI_VERSION_MAJOR': vinfo.major,
```

Comment:
Why do we need the major and minor version here? Just the version should be fine since we're pinning the full version. Will also simply the code a lot.

Reply:
Maybe I misunderstood your earlier comments with @cicorias. Don't we want at least patches to be updated to the latest patch version? 0.1.x not withstanding but I don't thing patches will introducing any breaking changes in the future, only bug/security fixes.

Reply:
Under the assumption that the project will follow semver, that's fine. However, this puts a burden on maintainers so I wonder how useful it will be. Having a varying version dependency in the dockerfile will also mean that builds are no longer deterministic which is an issue for concerns like auditing. IMO client updates should be handled via a service like [pyup](https://pyup.io/), [synk](https://snyk.io/) or similar and generating good changelogs as part of the release so that users know exactly what a new version brings.

Reply:
@c-w you raise a good point on having deterministic docker builds. We should add documentation that client update should be handled by the user (and to link to potential services like pyup, synk, etc. to help with that).